flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CommonCatalogOptions.java

```diff
@@ -41,4 +41,13 @@ public class CommonCatalogOptions {
      */
     public static final ConfigOption<String> CATALOG_TYPE =
             ConfigOptions.key("type").stringType().noDefaultValue();
+
+    public static final ConfigOption<String> TABLE_CATALOG_STORE_KIND =
+            ConfigOptions.key("table.catalog-store.kind")
+                    .stringType()
+                    .defaultValue("generic_in_memory")
```

Comment:
public static final String DEFAULT_XXX,  see GenericInMemoryCatalogFactoryOptions, you can defined this in GenericInMemoryCatalogStoreFactoryOptions